pkg/dashboard/resource/functiontemplate.go

```diff
@@ -17,24 +17,44 @@ limitations under the License.
 package resource
 
 import (
+	"io/ioutil"
 	"net/http"
+	"os"
 
 	"github.com/nuclio/nuclio/pkg/dashboard"
 	"github.com/nuclio/nuclio/pkg/dashboard/functiontemplates"
 	"github.com/nuclio/nuclio/pkg/errors"
 	"github.com/nuclio/nuclio/pkg/restful"
+
+	"github.com/fatih/structs"
+	"gopkg.in/yaml.v2"
+	"github.com/nuclio/nuclio-sdk-go"
+	"encoding/json"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"strings"
 )
 
 type functionTemplateResource struct {
 	*resource
 	functionTemplateRepository *functiontemplates.Repository
 }
 
+type RenderConfig struct {
+	Template string `json:"template,omitempty"`
+	Values map[string]interface{} `json:"values,omitempty"`
+}
+
 func (ftr *functionTemplateResource) OnAfterInitialize() error {
-	var err error
+	githuAPItoken := os.Getenv("NUCLIO_GITHUB_API_TOKEN")
+	supportedSuffixes := []string{".go", ".py"}
+
+	repoFetcher, err := functiontemplates.NewGithubFunctionTemplateFetcher("nuclio-templates", "ilaykav", "master", githuAPItoken, supportedSuffixes)
```

Comment:
1. this should be a part of the server, constructed outside of the resource. will show you how to do this tomorrow
2. the type here should be `functionTemplateFetcher`, not `repoFetcher`